### PR TITLE
feat: profile intent tests

### LINKED ISSUE #100: My Profile fragment intent Test

### PR DESCRIPTION
**Pull request by rjmah1**
#### Summary of changes/features added:
- 

#### Screenshots of UI changes (update with each new commit): 
- 

## Pull Request Checklist
- [ ] Unit test written?
- [x] (**For later in project**) Integration tests written?
- [x] Java docs written?

### CODE REVIEW

**Review by AdityaHarvi — CHANGES_REQUESTED** [review]
Delete the MyProfileTest (intent test) file and then we can replace that with this

**Review by AdityaHarvi — APPROVED** [review]

File: client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        solo.clickOnButton("Save");
+        assertTrue(solo.waitForText(testString, 1, 2000));
+    }
+
+    /**
+     * Closes the activity after each test
+     * @throws Exception
+     */
+    @After
+    public void tearDown() throws Exception{
```
Comment (SamLojpur) [inline client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java:163]: these don't seem to actually throw exception

File: client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    @Rule
+    public ActivityTestRule<LoginActivity> rule =
+            new ActivityTestRule<>(LoginActivity.class, true, true);
+
+    /**
+     * Runs before all tests and creates solo instance.
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception {
```
Comment (SamLojpur) [inline client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java:42]: these don't seem to actually throw exception

File: client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        assertTrue(solo.waitForText(testString, 1, 2000));
+        solo.clickOnButton("Cancel");
+        assertFalse(solo.waitForText(testString, 1, 2000));
+    }
+
+    /**
+     * Test to see if the fragment lets the user save the edit
+     */
+    @Test
+    public void editTextSaveTest() {
```
Comment (SamLojpur) [inline client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java:145]: are you reverting this after? once real saves come in you should revert it to bio

File: client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        solo.clickOnButton("Edit");
+        String testString = "This is a bio text test";
+        solo.clearEditText((EditText) solo.getView(R.id.bio_text));
+        assertFalse(solo.waitForText(testString, 1, 2000));
+        solo.enterText((EditText) solo.getView(R.id.bio_text), testString);
+        assertTrue(solo.waitForText(testString, 1, 2000));
+        solo.clickOnButton("Cancel");
+        assertFalse(solo.waitForText(testString, 1, 2000));
+    }
+
```
Comment (SamLojpur) [inline client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java:140]: could we add some tests asserting that the user info is "bio", "phone", "address". the third one probably wont pass for now lol

Comment (rjmah1) [reply client/app/src/androidTest/java/com/example/criengine/MyProfileFragmentTest.java:140]: idk how to write that kind of test without knowing what the "user" is in the db

**Review by SamLojpur — APPROVED** [review]

**Pull request merged**